static/app/components/deviceName.tsx

```diff
@@ -1,81 +1,82 @@
 import * as React from 'react';
+import * as Sentry from '@sentry/react';
 
+// Self reference to the module, so that we can mock a failed import in a test.
+import * as selfModule from 'sentry/components/deviceName';
 import {IOSDeviceList} from 'sentry/types/iOSDeviceList';
 
-export function deviceNameMapper(model: string, iOSDeviceList): string {
-  const modelIdentifier = model.split(' ')[0];
-  const modelId = model.split(' ').splice(1).join(' ');
-  const modelName = iOSDeviceList.generationByIdentifier(modelIdentifier);
-  return modelName === undefined ? model : modelName + ' ' + modelId;
+export function iOSDeviceNameMapper(
+  model: string | undefined,
+  iOSDeviceList: IOSDeviceList | null
+): string | null {
+  // If we have no model, render nothing
+  if (typeof model !== 'string') {
+    return null;
+  }
+
+  // If module has not loaded yet, render the unparsed model
+  if (iOSDeviceList === null) {
+    return model;
+  }
+
+  const [identifier, ...rest] = model.split(' ');
+
+  const modelName = iOSDeviceList.generationByIdentifier(identifier);
+  return modelName === undefined ? model : `${modelName} ${rest.join(' ')}`;
 }
 
-export async function loadDeviceListModule() {
+export async function loadiOSDeviceListModule() {
   return import('ios-device-list');
 }
 
-type Props = {
-  value: string;
+interface DeviceNameProps {
+  value: string | undefined;
   children?: (name: string) => React.ReactNode;
-};
+}
 
-type State = {
-  iOSDeviceList: IOSDeviceList | null;
-};
 /**
  * This is used to map iOS Device Names to model name.
  * This asynchronously loads the ios-device-list library because of its size
  */
-export default class DeviceName extends React.Component<Props, State> {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      iOSDeviceList: null,
+function DeviceName({value, children}: DeviceNameProps): React.ReactElement | null {
+  const [iOSDeviceList, setiOSDeviceList] = React.useState<IOSDeviceList | null>(null);
```

Comment:
Meh, I'm not a fan of either of the two tbh :D on one hand, it follows the convention, but on the other, it's mangling the name. I opted for this one because it makes the codebase slightly more searchable (e.g if you grep the codebase for iOS), you would have a hit here. I'm not sure which one of the two is better

Reply:
Actually, I found a better naming. Since the component is called DeviceName, we should make the loading functions named similarly. So we can rename the module loading functions not to be iOS specific. 

E.g. `function loadDeviceListModule(platform: 'iOS')`

Reply:
Just pushed the change, lmk what you think